src/containers/Topbar/Topbar.js

```diff
@@ -85,12 +93,48 @@ class Topbar extends React.PureComponent {
     this.handleCancel = this.handleCancel.bind(this);
     this.onAddressInputChange = this.onAddressInputChange.bind(this);
     this.onAddressDropdownClick = this.onAddressDropdownClick.bind(this);
-    this.getSelectedAddressObject = this.getSelectedAddressObject.bind(this);
     this.onCopyClicked = this.onCopyClicked.bind(this);
   }
 
   componentWillMount() {
-    this.props.listUnspent();
+    const { onGetBlockCount, listUnspent } = this.props;
+
+    (function startPoll() {
+      onGetBlockCount();
+      listUnspent();
+      setTimeout(startPoll, POOL_INTERVAL);
+    }());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {
+      onGetBlockCount, getBotBalance, selectedWalletAddress,
+    } = this.props;
+
+
+    // Call API to retrieve BOT balance if wallet addresses have changed
+    if (!_.isEqual(this.props.walletAddrs, nextProps.walletAddrs)) {
+      _.each(nextProps.walletAddrs, (addressObj) => {
+        const ownerAddress = addressObj.address;
+        let senderAddress;
+
+        // Determine an address that has enough qtum for gas to be a sender
+        if (addressObj.qtum >= MINIMAL_GAS_FEE) {
```

Comment:
what about if a user wants to `callcontract` (no gas needed) from an address that has no qtum?

Reply:
then he hits else case where we loop through his wallet addresses and find the first with enough qtum to pay for gas.

Reply:
What I meant to say is something like this case:
a user had Qtum before and he bet/voted on something, but now he has 0 qtum. but he is on the completed TopicEvent page and would like to check his `withdraw` amount (which costs no gas/qtum) amount on the from this address.  this logic would not allow him to do any free `callcontract` calls.